### DISCUSSION TITLE
Best practice for login required flows

### DISCUSSION THREAD

**Discussion by KrisLau** [mobile-dev-inc/Maestro#1053] [General]
So for 90% of my flows I would need the user to be logged in. Is the best practice to `clearState` & run a login flow then the actual test OR should I be running the login test manually then running the other tests after? If it's the latter, is it possible to do so from the config file? Also should I be launching the app each time?

**Comment by axelniklasson** [comment]
Hi there @KrisLau! Best practice would indeed be to use `clearState` and extracting out a login Flow that you would run before your actual Flows, so there is no dependency between your Flows. That also allows you to run the Flows in parallel in CI like when running on [Maestro Cloud](https://cloud.mobile.dev/).
